src/core/components/Dialog.tsx

```diff
@@ -13,6 +14,7 @@ import {
 type DialogProps = {
   open: boolean;
   onClose: (value: any) => void;
+  onClick?: MouseEventHandler<HTMLDivElement>;
```

Comment:
Alternatively we might want to expose/extend the props of the underlying HTML element